igcollect/jmx.py

```diff
@@ -19,6 +19,7 @@ def parse_args():
     parser.add_argument('--ports', default=[], nargs='+')   # XXX Deprecated
     parser.add_argument('--names', default=[], nargs='+')   # XXX Deprecated
     parser.add_argument('--thread-prefixes', default=[], nargs='+')
+    parser.add_argument('--metric-file', default='etc/jmx2graphite/metrics', nargs='?')
```

Comment:
Specifying nargs with `?` will also allow passing the argument without value e.g. `--metric-file` instead of `--metric-file /etc/something`.

You may want to specify const or use a different nargs value. You can consult the [docs](https://docs.python.org/3/library/argparse.html?highlight=argument%20parser#nargs)

Reply:
to my understanding is passing it without argument will always produce the default which is correct here

Reply:
if its emtpy it now should now throw an error jmx.py: error: argument --metric-file: expected one argument